engine/table/src/main/java/io/deephaven/engine/util/LayoutHintBuilder.java

```diff
@@ -105,9 +108,63 @@ static AutoFilterData fromString(String string) {
         }
     }
 
+    private static class ColumnGroup {
+
+        String name;
+        final List<String> children;
+        Color color;
+
+        ColumnGroup(String name, List<String> children, Color color) {
+            NameValidator.validateColumnName(name);
+            children.forEach(c -> NameValidator.validateColumnName(c));
+
+            this.name = name;
+            this.children = children;
+
+            if (color != null) {
+                this.color = color;
+            }
+        }
+
+        @NotNull
+        String serialize() {
+            StringBuilder sb = new StringBuilder("name:" + name);
+
+            sb.append("::children:" + StringUtils.joinStrings(children, ","));
+            if (color != null) {
+                sb.append("::color:" + "#" + Integer.toHexString(color.javaColor().getRGB()).substring(2));
+            }
+            return sb.toString();
+        }
+
+        /**
+         * Convert a string of the format defined by {@link #serialize()} into a proper ColumnGroup object
+         *
+         * @param string the string to parse
+         * @return a ColumnGroup instance
+         */
+        @NotNull
+        static ColumnGroup fromString(String string) {
+            final Map<String, String> options = Arrays.stream(string.split("::"))
+                    .map(option -> option.split(":"))
+                    .collect(Collectors.toMap(parts -> parts[0], parts -> parts.length == 2 ? parts[1] : ""));
```

Comment:
why empty string instead of null? explicit null makes your `if` below simpler, and i dont think either empty string nor null are safe for name, or children?

Reply:
I think I was just going off of the `JsLayoutHints.parse` call. Explicit null will work here